cranelift/codegen/src/ir/instructions.rs

```diff
@@ -284,6 +285,39 @@ impl InstructionData {
         }
     }
 
+    /// Return the value of an immediate if the instruction has one or `None` otherwise. Only
+    /// immediate values are considered, not global values, constant handles, condition codes, etc.
+    pub fn imm_value(&self) -> Option<DataValue> {
+        match self {
+            &InstructionData::UnaryBool { imm, .. } => Some(DataValue::from(imm)),
+            // 8-bit.
+            &InstructionData::BinaryImm8 { imm, .. }
+            | &InstructionData::BranchTableEntry { imm, .. } => Some(DataValue::from(imm as i8)), // Note the switch from unsigned to signed.
+            // 32-bit
+            &InstructionData::UnaryIeee32 { imm, .. } => Some(DataValue::from(imm)),
+            &InstructionData::HeapAddr { imm, .. } => {
+                Some(DataValue::from(Into::<u32>::into(imm) as i32)) // Note the switch from unsigned to signed.
```

Comment:
Here and on line 295 I had to switch unsigned to signed--there are other ways to do this (bump up the bit width, e.g.) but I'm not really sure it matters (users of this probably know what to do?)

Reply:
For clarity, maybe `let imm: u32 = imm.into();` then we can write `DataValue::from(imm as i32)`?